### ISSUE TITLE
WL4200S not updating temperature values

### ISSUE THREAD

**Issue by levinse** [claudegel/sinope-130#87]
Not sure if this is the correct place for this issue, but I have multiple WL4200S sensors and also a WL4200 sensor. The WL4200 is updating its temperature values frequently (multiple times per hour, if there are changes) in Home Assistant but the WL4200S sensors never change their values. The WL4200S sensors otherwise work fine (e.g. they trigger the leak sensor and the Last Seen value on the device page is generally only a few minutes old).

Anything specific I can try to troubleshoot this further, or if this isn't the right place for this, where would I open the issue?

EDIT: I'm using Conbee II and not neviweb

**Comment by claudegel** [comment]
Yes you are at the wright place. Il also have plenty of wl4200 so I can help. This weekend I'm grandfather on duty so I'll check tonight when m'y Little children battery are off😄

**Comment by claudegel** [comment]
Please tell me if your WL4200S are properly detected in zha-quirks. Especially for the line underlined in red below
![WL4200](https://user-images.githubusercontent.com/4268938/162585085-f90dc753-f8b3-4b1a-a4b5-bd66a0dab80e.png)

Then send me your zigbee information for the WL4200S like below
![WL4200-1](https://user-images.githubusercontent.com/4268938/162585171-58d63cbb-edf1-48e1-bf15-4e40c168b6e8.png)

**Comment by levinse** [comment]
> This weekend I'm grandfather on duty so I'll check tonight when m'y Little children battery are off

I'm on father duty so I have the same here :)

> Please tell me if your WL4200S are properly detected in zha-quirks. Especially for the line underlined in red below. 

It looks slightly different from yours for the line in red and also the firmware version is different:

![Screenshot from 2022-04-09 16-00-55](https://user-images.githubusercontent.com/14372217/162589853-b5fc0479-a908-42d5-a7c2-dfa8b5e36d17.png)

> Then send me your zigbee information for the WL4200S like below

```
{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=1, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4508, maximum_buffer_size=71, maximum_incoming_transfer_size=43, server_mask=10752, maximum_outgoing_transfer_size=43, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": 260,
      "device_type": "0x0402",
      "in_clusters": [
        "0x0000",
        "0x0001",
        "0x0003",
        "0x0020",
        "0x0402",
        "0x0500",
        "0x0b05",
        "0xff01"
      ],
      "out_clusters": [
        "0x0003",
        "0x0019"
      ]
    }
  },
  "manufacturer": "Sinope Technologies",
  "model": "WL4200S",
  "class": "zhaquirks.sinope.sensor.SinopeTechnologiesSensor2"
}
```

**Comment by claudegel** [comment]
I've switched my WL4200S sensor from Neviwev to ZHA and it is displaying temperature correctly
![WL4200S](https://user-images.githubusercontent.com/4268938/162597317-26af8053-9d24-477e-a31a-b1e570198f01.png)
You can see that temperature is 15.5oC. I've move it to higher temperature location and I can see the temperature changes. The only thing is, mine is a first generation WL4200S sensor and you have a 2e generation sensor.
Mine is reported as Quirk: sensor.SinopeTechnologiesSensor and yours is reported as Quirks: zhaquirks.sinope.sensor.SinopeTechnilogiesSensor2. This is normal as you are using ZHA with the sinope quirks for sensor and I'm using  a local quirks that I use to test ZHA sinope quirks update
Your sensor is correctly detected by zha-quirks and the sinope quirks for sensor 2e generation is used. I can see on the picture you sent that temperature is 18.5oC. If you place your sensor in a higher temperature location, does the temperature change ?

**Comment by claudegel** [comment]
If you want to test latest quirks locally on your HA server you can check my https://github.com/claudegel/sinope-zha. This is the latest version of the sinope quirks that are updated before they appear in the ZHA version in HA.

**Comment by levinse** [comment]
Thanks Claude! No changes to the temperature. I'll try your quirks version.

**Comment by claudegel** [comment]
Your sensor id correctly detected by ZHA-Quirks so Idon't think it is software dependant. Since the temperature is displayed then the cluster is published by the sensor. I suspect that the temperature sensor is broken or frozen. You can also try to remove the battery for couple hour then re plug the battery to see if the temperature will be updated. If you have a GT130 you can connect your sensor on Neviweb to see if there will be a firmware update this is dome automatically in 24-48 hours. You can also make sure the freeze protect is activated.  Without freeze protection active it is possible that the temperature is not updated. I'm gonna sniff the zigbee line for my WL4200S to see if there is other specific cluster that we need to add to the quirks.

**Comment by levinse** [comment]
Thanks. I actually have the same behaviour on 3 different WL4200S so I don't think it's one specific sensor that's broken, but I only have one WL4200 so perhaps that one is accidentally working :)

**Comment by levinse** [comment]
I don't have a GT130 unfortunately

**Comment by claudegel** [comment]
Hi,
I've found that with zha-toolkit you can do a cluster scan of your device and see if there are reporting available for the cluster/attribute that should report the temperature.
Then with zha-toolkit you can configure the reporting for the temperature. 
As you never connect your sensor to Neviweb it is possible that the reporting is just not activated.
Since few week it is now possible to see your manufacturer cluster 0xff01 and attribute in zha-toolkit and you will find information on my claudegel/sinope-zha on how to enable reporting for cluster/attribute that are configured to do reporting.
Hope this can help
